e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js

```diff
@@ -507,6 +451,102 @@ describe("scenarios > dashboards > filters > auto apply", () => {
+            name: "Full-app embedding dashboard",
+            auto_apply_filters: false,
+          },
+        });
+        cy.get("@dashboardId").then(dashboardId => {
+          visitFullAppEmbeddingUrl({
+            url: `/dashboard/${dashboardId}`,
+            qs: { side_nav: false, logo: false },
+          });
+        });
+        cy.findByDisplayValue("Full-app embedding dashboard").should(
+          "be.visible",
+        );
+        // Ensure that we're viewing the dashboard in full-app embedding mode, since `logo` is a full-app embedding parameter.
+        cy.findByTestId("main-logo").should("not.exist");
+
+        dashboardParametersContainer().button("Apply").should("not.exist");
+        filterWidget().findByText("Category").click();
+        popover().within(() => {
+          cy.findByText("Widget").click();
+          cy.button("Add filter").click();
+        });
+        getDashboardCard().findByText("Rows 1-5 of 200").should("be.visible");
+        dashboardParametersContainer()
+          .button("Apply")
+          .should("be.visible")
+          .click();
+        getDashboardCard().findByText("Rows 1-5 of 54").should("be.visible");
+      });
+
+      it("should display a toast when a dashboard takes longer than 15s to load", () => {
+        createDashboard();
+        // Not sure why I need to pass a date in this case, but it doesn't work without it.
+        cy.clock(Date.now());
+        openSlowFullAppEmbeddingDashboard({ [FILTER.slug]: "Gadget" });
+        cy.tick(TOAST_TIMEOUT);
+        cy.wait("@cardQuery");
+        undoToast().within(() => {
+          cy.findByText(TOAST_MESSAGE).should("be.visible");
+          cy.button("Turn off").click();
+          cy.wait("@updateDashboard");
+        });
+
+        // In embedding we'll load bookmark after the dashboard is loaded, it's the opposite in normal app because bookmark is cached from somewhere else.
+        // And somehow, dashboard card query will be completed before the dashboard even start to load, and in entity loader it uses `setTimeout`,
+        // so to make sure callback in `setTimeout` is called, we need to advance the clock using cy.tick().
+        cy.tick();
+
+        toggleDashboardInfoSidebar();
+        rightSidebar()
+          .findByLabelText("Auto-apply filters")
+          .should("not.be.checked");
+        filterWidget().findByText("Gadget").should("be.visible");
+
+        // Card height isn't updated because we're mocking clock, that's it's only showing 1 result per page.
+        getDashboardCard().findByText("Rows 1-1 of 53").should("be.visible");
+
+        // card result should be updated after manually updating the filter
+        filterWidget().icon("close").click();
+        dashboardParametersContainer()
+          .button("Apply")
+          .should("be.visible")
+          .click();
+
+        // Card height isn't updated because we're mocking clock, that's it's only showing 1 result per page.
```

Comment:
I don't understand the second part of this comment (there are 3 comments like that).

Reply:
I meant to typed
> that's **why** it's only showing 1 result per page

Here's the result from that test that uses `cy.clock()`
![image](https://github.com/metabase/metabase/assets/1937582/b80243dd-1d76-4cd4-b064-0d76d96d1cf7)

If we don't mock the clock aka using `cy.clock()`
![image](https://github.com/metabase/metabase/assets/1937582/6ba063a6-8e25-4427-9488-b9ecf0f3bc54)

The reason is that (I guess), this is the work of `ExplicitSize` HoC which we use to wrap visualization in dashcards on dashboards. It has various refresh modes which internally probably use some kind of `setTimeout` or `setInterval`. That's why it won't refresh the layout.

https://github.com/metabase/metabase/blob/6d1bf160dbffbb07b5efa24589ea349f63171bd5/frontend/src/metabase/components/ExplicitSize.jsx#L12-L17

[This is a PR](https://github.com/metabase/metabase/pull/22094), that introduced this change to `ExplicitSize` HoC for optimization. (`ExplicitSize` is a HoC used to handle resizing our visualizations)

Reply:
Improve clarity in the comment in https://github.com/metabase/metabase/pull/31898/commits/7d24274f4a0d5ca6c942134b88f7b26331f272e5

It's more verbose, but we don't need to repeat that comment too many times (since it's specifically to only full-app embeddings from my testing), so I don't think it's that bad.